### PR TITLE
MCR-1865 fix auto store creation on read operations

### PR DESCRIPTION
**Pull request by Mewel**
-add readOnly option
-use Path API in MCRXMLMetadataManager

### CODE REVIEW

**Comment by MyCoRe-Travis** [comment]
Something gone wrong on a8131caebcccfc1b8c8c9ea5dbc89c06817f54cd! 
 Please Check https://github.com/MyCoRe-Travis/test_artifacts/tree/3297-pull_request-592

File: mycore-base/src/main/java/org/mycore/datamodel/common/MCRXMLMetadataManager.java [hunk 1]
```diff
@@ -216,42 +219,103 @@ private synchronized void closeCreatedStores() {
...
+     * @param path the path to check
+     * @param type metadata store type
      */
-    private void checkDir(File dir, String type) {
-        if (!dir.exists()) {
+    private void checkDir(Path path, String type) {
+        if (!Files.exists(path)) {
             try {
-                boolean created = dir.mkdirs();
+                boolean created = Files.exists(Files.createDirectories(path));
```
Comment (yagee-de) [inline mycore-base/src/main/java/org/mycore/datamodel/common/MCRXMLMetadataManager.java]: misleading name for variable

File: mycore-base/src/main/java/org/mycore/datamodel/common/MCRXMLMetadataManager.java [hunk 2]
```diff
@@ -216,42 +219,103 @@ private synchronized void closeCreatedStores() {
...
                 if (!created) {
-                    String msg = "Unable to create metadata store " + type + " directory " + dir.getAbsolutePath();
+                    String msg = "Unable to create metadata store " + type + " directory " + path.toAbsolutePath();
```
Comment (yagee-de) [inline mycore-base/src/main/java/org/mycore/datamodel/common/MCRXMLMetadataManager.java]: msg is misleading, too

File: mycore-base/src/main/java/org/mycore/datamodel/common/MCRXMLMetadataManager.java [hunk 3]
```diff
@@ -216,42 +219,103 @@ private synchronized void closeCreatedStores() {
...
             }
-            if (!dir.canWrite()) {
-                String msg = "Metadata store " + type + " directory " + dir.getAbsolutePath() + " is not writeable";
+            if (!Files.isWritable(path)) {
+                String msg = "Metadata store " + type + " directory " + path.toAbsolutePath() + " is not writeable";
                 throw new MCRConfigurationException(msg);
             }
-            if (!dir.isDirectory()) {
-                String msg = "Metadata store " + type + " " + dir.getAbsolutePath() + " is a file, not a directory";
+            if (!Files.isDirectory(path)) {
```
Comment (yagee-de) [inline mycore-base/src/main/java/org/mycore/datamodel/common/MCRXMLMetadataManager.java]: isDirectory should be checked first for other "directory" exceptions listed before...

File: mycore-base/src/main/java/org/mycore/datamodel/common/MCRXMLMetadataManager.java [hunk 4]
```diff
@@ -295,67 +360,60 @@ private void setupStore(String project, String objectType, String configPrefix)
...
+            throw new MCRPersistenceException(MessageFormat.format(
+                    "Path does not exists ''{0}'' to set up store for project ''{1}'' and objectType ''{2}'' "
+                            + "and config prefix ''{3}''. We are not willing to create it for an read only operation.",
+                    path.toAbsolutePath(), project, objectType, configPrefix));
+        }
+        try {
+            if (!Files.exists(Files.createDirectories(path))) {
+                throw new FileNotFoundException(path.toAbsolutePath() + " does not exists.");
+            }
+        } catch (Throwable t) {
```
Comment (yagee-de) [inline mycore-base/src/main/java/org/mycore/datamodel/common/MCRXMLMetadataManager.java]: change Throwable to Exception

**Review by yagee-de — CHANGES_REQUESTED** [review]

**Review by yagee-de — APPROVED** [review]

**Pull request merged**